docs/source/development/style-guide.rst

```diff
@@ -0,0 +1,392 @@
+<https://numpydoc.readthedocs.io/en/latest/format.html>`_ conventions.
+
+The following are further recommendations:
+
+* Each module should have at minimum a description and a ``Use`` section.
+* Each function/method should have at minimum a description, ``Parameters`` (if necessary), and ``Returns`` (if
+  necessary) sections.
+
+When a new module is added to the repository, make sure to add the module to the appropriate ``docs/source/*.rst`` file,
+so that the API documentation can be part of the automated documentation build.
+
+
+.. _poetry-environment:
+Poetry Environment
+------------------
+
+Poetry is used for dependency management within this project. To update dependencies, you can either update
+``pyproject.toml`` manually, or use ``poetry add <dependency>``.
+
+If you do add dependencies, please make sure you define the version numbers mindfully. The best way to do this is to use
+`caret notation <https://python-poetry.org/docs/dependency-specification/#caret-requirements>`_ to allow for minor
+version updates. For example, if you have tested the code using numpy version ``1.24.2``, the best way to specify the
+dependency is to allow for updates to the right most version number:
+
+::
+
+    numpy = "^1.24"
+
+This will allow for patches for security reasons without upgrading to ``1.25`` or beyond. Use your best judgement for
+when packages can upgrade automatically, but try and avoid specifying only one specific version unless it is absolutely
+required.
+
+The ``poetry.lock`` file contains the existing dependencies for the project. These are the dependencies that you should
+install to ensure you're getting the accepted versions for every package. If the ``poetry.lock`` file is not up-to-date,
+there is an automatic process within the repo to update ``poetry.lock`` to match ``pyproject.toml``. To resolve any
+dependency issues before the pre-commit hook, users should run ``poetry lock`` and resolve any dependency problems ahead
+of time. The ``poetry.lock`` file should be part of the files committed in this case.
+
+
+.. _security:
+Security
+--------
+
+The following items should never be committed in the ``imap_processing`` source code or GitHub issues/pull requests:
+
+* Account credentials of any kind (e.g. database usernames/passwords, AWS credentials, etc.)
+* Internal directory structures or filepaths
+* Machine names
+* Proprietary data
+
+If ``imap_processing`` code needs access to this information, it should be stored in a configuration file that is not
+part of the repository.
+
+
+.. _naming-conventions:
+Naming Conventions
+------------------
+
+The ``imap_processing`` repo shall adhere to specific naming conventions to allow for more consistency:
+
+* Repositories shall have branches named ``main`` and ``dev``, instead of ``master`` and ``development``
+* Repositories shall be named with underscores instead of hyphens (e.g. ``imap_processing`` instead of
+  ``imap-processing``)
```

Comment:
Here is where I added a few items based on what we decided in a previous Slack convo